library/ecdh.c

```diff
@@ -278,53 +414,114 @@ int mbedtls_ecdh_get_params( mbedtls_ecdh_context *ctx, const mbedtls_ecp_keypai
 }
 
 /*
- * Setup and export the client public value
+ * Get parameters from a keypair
  */
-int mbedtls_ecdh_make_public( mbedtls_ecdh_context *ctx, size_t *olen,
-                      unsigned char *buf, size_t blen,
-                      int (*f_rng)(void *, unsigned char *, size_t),
-                      void *p_rng )
+int mbedtls_ecdh_get_params( mbedtls_ecdh_context *ctx,
+                             const mbedtls_ecp_keypair *key,
+                             mbedtls_ecdh_side side )
+{
+    int ret;
+
+    if( ctx == NULL )
+        return( MBEDTLS_ERR_ECP_BAD_INPUT_DATA );
+
+    if( ( ret = mbedtls_ecdh_setup( ctx, key->grp.id ) ) != 0 )
```

Comment:
The code has changed from `mbedtls_ecp_group_copy( &ctx->grp, &key->grp )` to `mbedtls_ecp_group_load( &ctx->grp, key->grp.id )`. This is correct only if there is a single group with any given ID. This applies to our own code, but might some users have added their own groups? I think that mandating that groups have distinct IDs is a reasonable requirement, but was it already a requirement before for ECDH? Is it properly documented?

Reply:
I have added clarification about this requirement.